### ISSUE TITLE
Unable to install PlotlyJS on Julia 0.6 on Mac

### ISSUE THREAD

**Issue by dpsanders** [sglyon/PlotlyJS.jl#139]
Starting from a basically fresh .julia, I get

```
julia> Pkg.clone("PlotlyJS")
INFO: Cloning PlotlyJS from git://github.com/sglyon/PlotlyJS.jl.git
INFO: Computing changes...
Pkg.INFO: No packages to install, update or remove
INFO: Package database updated

julia> Pkg.build("PlotlyJS")
INFO: Building HttpParser
INFO: Building Homebrew
INFO: Building MbedTLS
INFO: Building SpecialFunctions
INFO: Building PlotlyJS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  979k    0  979k    0     0   437k      0 --:--:--  0:00:02 --:--:--  437k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2285k  100 2285k    0     0  3491k      0 --:--:-- --:--:-- --:--:-- 3489k

WARNING: deprecated syntax "_ as an rvalue around /Users/davidsanders/.julia/v0.6/PlotlyJS/deps/make_schema_docs.jl:22".
==============================[ ERROR: PlotlyJS ]===============================

LoadError: LoadError: MethodError: Cannot `convert` an object of type String to an object of type __anon__.PlotlyJSSchemaDocsGenerator.SchemaAttribute
This may have arisen from a call to the constructor __anon__.PlotlyJSSchemaDocsGenerator.SchemaAttribute(...),
since type constructors fall back to convert methods.
while loading /Users/davidsanders/.julia/v0.6/PlotlyJS/deps/make_schema_docs.jl, in expression starting on line 227
while loading /Users/davidsanders/.julia/v0.6/PlotlyJS/deps/build.jl, in expression starting on line 6

================================================================================
ERROR: UndefVarError: __anon__ not defined
Stacktrace:
 [1] deserialize_module(::SerializationState{IOStream}) at ./serialize.jl:754
 [2] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:695
 [3] deserialize(::SerializationState{IOStream}) at ./serialize.jl:634
 [4] deserialize_datatype(::SerializationState{IOStream}, ::Bool) at ./serialize.jl:968
 [5] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:674
 [6] deserialize(::SerializationState{IOStream}) at ./serialize.jl:634
 [7] ntuple(::Base.Serializer.##1#2{SerializationState{IOStream}}, ::Int64) at ./tuple.jl:108
 [8] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:664
 [9] deserialize(::SerializationState{IOStream}, ::DataType) at ./serialize.jl:1075
 [10] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:687
 [11] deserialize(::SerializationState{IOStream}, ::DataType) at ./serialize.jl:1075
 [12] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:687
 [13] deserialize(::SerializationState{IOStream}, ::DataType) at ./serialize.jl:1075
 [14] handle_deserialize(::SerializationState{IOStream}, ::Int32) at ./serialize.jl:687
 [15] (::Base.Pkg.Entry.##54#55{Dict{Any,Any}})(::IOStream) at ./pkg/entry.jl:639
 [16] open(::Base.Pkg.Entry.##54#55{Dict{Any,Any}}, ::String, ::String) at ./iostream.jl:152
 [17] build!(::Array{String,1}, ::Dict{Any,Any}, ::Set{Any}) at ./pkg/entry.jl:636
 [18] build(::Array{String,1}) at ./pkg/entry.jl:650
 [19] (::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}})() at ./pkg/dir.jl:36
 [20] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#build,Tuple{Array{String,1}}}, ::String) at ./file.jl:70
 [21] #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{String,1}, ::Vararg{Array{String,1},N} where N) at ./pkg/dir.jl:36
 [22] build(::String, ::Vararg{String,N} where N) at ./pkg/pkg.jl:254

```

The same happened with `Pkg.add("PlotlyJS")`.

**Comment by babaq** [comment]
same on windows 7

**Comment by sglyon** [comment]
Thank you both for reporting. I just came up with a fix and will push/publish it soon

**Comment by sglyon** [comment]
Right now this is being blocked by issues with home brew on OS X high Sierra. Hopefully that is resolved soon and this will be fixed.

**Comment by v-i-s-h** [comment]
I'm in OSX Yosemite and is experiencing the same problem.

**Comment by rdeits** [comment]
I'm seeing this on Linux too. Downgrading PlotlyJS all the way to `v0.5.2` fixed the problem for now.

**Comment by sglyon** [comment]
Wow, ok. thanks for the info @rdeits .

Unfortunately I won't be able to approach this until my dev setup is properly so unfortunately we will just have to sit tight and hope the home-brew + high Sierra issues are sorted out quickly.

**Comment by rdeits** [comment]
Could you point me to the homebrew/High Sierra issue?

**Comment by sglyon** [comment]
From the issue list, it looks like there are now two of them: https://github.com/JuliaPackaging/Homebrew.jl/issues

**Comment by v-i-s-h** [comment]
One work around is to run 
```run(`install_name_tool -change @rpath/libiomp5.dylib @loader_path/libiomp5.dylib $(Pkg.dir("Conda", "deps/usr/lib/libmkl_intel_thread.dylib"))`)```

mentioned in https://github.com/JuliaPy/PyPlot.jl/issues/315#issuecomment-335661614

**Comment by ghuba** [comment]
Same problem here. 

You may want to have a look at this to see if it helps

https://github.com/staticfloat/homebrew-juliadeps/issues/119

**Comment by sglyon** [comment]
Thanks, that worked for me. Can anyone else confirm?

**Comment by ghuba** [comment]
Cairo was installed successfully. I still got this

```
INFO: Building PlotlyJS
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  979k    0  979k    0     0  1316k      0 --:--:-- --:--:-- --:--:-- 1316k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2287k  100 2287k    0     0  1336k      0  0:00:01  0:00:01 --:--:-- 1336k

WARNING: deprecated syntax "_ as an rvalue around /Users/.../.julia/v0.6/PlotlyJS/deps/make_schema_docs.jl:22".
=====================================================================================================================================================[ ERROR: PlotlyJS ]=====================================================================================================================================================

LoadError: LoadError: MethodError: Cannot `convert` an object of type String to an object of type __anon__.PlotlyJSSchemaDocsGenerator.SchemaAttribute
This may have arisen from a call to the constructor __anon__.PlotlyJSSchemaDocsGenerator.SchemaAttribute(...),
since type constructors fall back to convert methods.
while loading /Users/.../.julia/v0.6/PlotlyJS/deps/make_schema_docs.jl, in expression starting on line 227
while loading /Users/.../.julia/v0.6/PlotlyJS/deps/build.jl, in expression starting on line 6

=============================================================================================================================================================================================================================================================================================================================
ERROR: UndefVarError: __anon__ not defined
```

You know how to fix this?

**Comment by sglyon** [comment]
Yes, I actually just pushed a fix to master for that. 

```julia
Pkg.checkout("PlotlyJS"); Pkg.build("PlotlyJS")
```

should fix that

**Comment by pwl** [comment]
That also fixed the issue on Linux.  Thanks!

**Comment by ghuba** [comment]
It's semi-working. I have two Julia 0.6 on my machine:

https://julialang.org/downloads/

JuliaPro

and Julia for macOS

https://julialang-s3.julialang.org/bin/osx/x64/0.6/julia-0.6.0-osx10.7+.dmg

With JuliaPro, I was able to install PlotlyJS. I add to do a few checkout/build on a couple of packages. But the process eventually completed. I got some warnings for Package HttpServer and AtomShell but no error.

With Julia for macOS, I was able to build PlotlyJS but got an error with IJulia and ZMQ (I deleted .julia and restarted with a clean install; JuliaPro does not use .julia as far as I can tell).

https://github.com/JuliaInterop/ZMQ.jl/issues/157

**Comment by fipelle** [comment]
Still having some troubles on OSX 10.12.5 with Julia 0.5.2. 

================================================================================[ ERROR: PlotlyJS ]=================================================================================

LoadError: LoadError: syntax: extra token "SchemaAttribute" after end of expression
while loading /Users/.../.julia/v0.5/PlotlyJS/deps/make_schema_docs.jl, in expression starting on line 11
while loading /Users/.../.julia/v0.5/PlotlyJS/deps/build.jl, in expression starting on line 6

**Comment by sglyon** [comment]
The master branch of the repo only works on julia >= 0.6.0

Can you not `Pkg.add("PlotlyJS")`given your OSX and julia versions?

**Comment by fipelle** [comment]
Unfortunately not, I get the `"ERROR: UndefVarError: __anon__ not defined"` error. It works when I downgrade it to `v0.5.2`.

**Comment by ghuba** [comment]
Did you try to update homebrew as described in

staticfloat/homebrew-juliadeps#119

Does that help? This is for Julia 0.6.

**Comment by fipelle** [comment]
Yes, I tried but it doesn't work. At some point I will upgrade Julia to 0.6, but not having Gallium is an problematic.

**Comment by sglyon** [comment]
@dpsanders , @babaq , @rdeits, @ghuba 

New releases for both Julia 0.5 and >= v0.6.0 were officially merged into metadata yesterday. Can you please confirm that they install and build properly for you?

Thank you

**Comment by ghuba** [comment]
Beautiful. I did a successful reinstall from scratch (deleted .julia). Everything seems to work.

**Comment by babaq** [comment]
latest version build success in Julia 0.6 on windows 10
